assign_rights/assemble.py

```diff
@@ -6,9 +8,12 @@
 class RightsAssembler(object):
     """docstring for RightsCalculator"""
 
-    def retrieve_rights(self):
+    def retrieve_rights(self, item, shell_list):
```

Comment:
```suggestion
    def retrieve_rights(self, item):
```
I'd append to `shell_list` in the `for` loop below rather than passing it in here.